docs/content/guides/dialog/dialog/dialog.md

```diff
@@ -0,0 +1,341 @@
+---
+id: 1xi1xek4
+title: Dialog
+metaTitle: Dialog - JavaScript Data Grid | Handsontable
+description: Display modal dialogs, alerts, loading indicators, and notifications to enhance user interaction and provide feedback in your data grid application.
+permalink: /dialog
+canonicalUrl: /dialog
+tags:
+  - dialog
+  - modal
+  - popup
+  - alert
+  - confirm
+  - prompt
+react:
+  id: 55z3zjaz
+  metaTitle: Dialog - React Data Grid | Handsontable
+angular:
+  id: vq1llzfz
+  metaTitle: Dialog - Angular Data Grid | Handsontable
+searchCategory: Guides
+category: Dialog
+---
+
+# Dialog
+
+Display modal dialogs, alerts, loading indicators, and notifications to enhance user interaction and provide feedback in your data grid application.
+
+[[toc]]
+
+## Overview
+
+The Dialog plugin provides a modal dialog system for Handsontable that allows you to display custom content in modal dialogs that overlay the table. This is useful for showing notifications, error messages, loading indicators, or any other interactive content that requires user attention.
+
+The dialog system is designed to be flexible and customizable, supporting various content types including plain text, HTML, and DOM elements. It also provides options for styling, animations, and user interaction controls.
+
+## Basic configuration
+
+To enable the Dialog plugin, set the [`dialog`](@/api/options.md#dialog) option to `true` or provide a configuration object.
+
+::: only-for javascript
+
+::: example #example1 --js 1 --ts 2
+
+@[code](@/content/guides/dialog/dialog/javascript/example1.js)
+@[code](@/content/guides/dialog/dialog/javascript/example1.ts)
+
+:::
+
+:::
+
+::: only-for react
+
+::: example #example1 :react --js 1 --ts 2
+
+@[code](@/content/guides/dialog/dialog/react/example1.jsx)
+@[code](@/content/guides/dialog/dialog/react/example1.tsx)
+
+:::
+
+:::
+
+::: only-for angular
+
+::: example #example1 :angular --ts 1 --html 2
+
+@[code](@/content/guides/dialog/dialog/angular/example1.ts)
+@[code](@/content/guides/dialog/dialog/angular/example1.html)
+
+:::
+
+:::
+
+## Content types
+
+The dialog supports multiple content types including plain text, HTML strings, and DOM elements.
+
+### Plain text content
+
+::: only-for javascript
+
+::: example #example2 --js 1 --ts 2
+
+@[code](@/content/guides/dialog/dialog/javascript/example2.js)
+@[code](@/content/guides/dialog/dialog/javascript/example2.ts)
+
+:::
+
+:::
+
+::: only-for react
+
+::: example #example2 :react --js 1 --ts 2
+
+@[code](@/content/guides/dialog/dialog/react/example2.jsx)
+@[code](@/content/guides/dialog/dialog/react/example2.tsx)
+
+:::
+
+:::
+
+::: only-for angular
+
+::: example #example2 :angular --ts 1 --html 2
+
+@[code](@/content/guides/dialog/dialog/angular/example2.ts)
+@[code](@/content/guides/dialog/dialog/angular/example2.html)
+
+:::
+
+:::
+
+### HTML content
+
+::: only-for javascript
+
+::: example #example3 --js 1 --ts 2
```

Comment:
I'd extend the demo and attach the event listener to the "Click me" button to show that the clicking actually works and changes the content of the dialog WDYT?